packages/services/api/src/modules/organization/providers/personal-access-tokens.ts

```diff
@@ -0,0 +1,409 @@
+
+    edges = edges.slice(0, limit);
+
+    return {
+      edges,
+      pageInfo: {
+        hasNextPage,
+        hasPreviousPage: cursor !== null,
+        get endCursor() {
+          return edges[edges.length - 1]?.cursor ?? '';
+        },
+        get startCursor() {
+          return edges[0]?.cursor ?? '';
+        },
+      },
+    };
+  }
+
+  /**
+   * Implementation for finding a personal access token from the PG database.
+   * It is a static function, so we can use it for the personal access tokens cache.
+   */
+  static findById(deps: { pool: CommonQueryMethods; logger: Logger }) {
+    return async function findByIdImplementation(personalAccessTokenId: string) {
+      deps.logger.debug(
+        'Resolve organization access token by id. (personalAccessTokenId=%s)',
+        personalAccessTokenId,
+      );
+
+      if (isUUID(personalAccessTokenId) === false) {
+        deps.logger.debug(
+          'Invalid UUID provided. (personalAccessTokenId=%s)',
+          personalAccessTokenId,
+        );
+        return null;
+      }
+
+      const data = await deps.pool.maybeOne<unknown>(sql`
+        SELECT
+          ${personalAccessTokenFields}
+        FROM
+          "personal_access_tokens"
+        WHERE
+          "id" = ${personalAccessTokenId}
+        LIMIT 1
+      `);
+
+      if (data === null) {
+        deps.logger.debug(
+          'Personal access token not found. (organizationAccessTokenId=%s)',
+          personalAccessTokenId,
+        );
+        return null;
+      }
+
+      const result = PersonalAccessTokenModel.parse(data);
+
+      deps.logger.debug(
+        'Personal access token found. (organizationAccessTokenId=%s)',
+        personalAccessTokenId,
+      );
+
+      return result;
+    };
+  }
+}
+
+const personalAccessTokenFields = sql`
+  "id"
+  , "organization_id" AS "organizationId"
+  , "user_id" AS "userId"
+  , to_json("created_at") AS "createdAt"
+  , "title"
+  , "description"
+  , "permissions"
+  , "assigned_resources" AS "assignedResources"
+  , "first_characters" AS "firstCharacters"
+  , "hash"
+`;
+
+const PersonalAccessTokenModel = z
+  .object({
+    id: z.string().uuid(),
+    organizationId: z.string().uuid(),
+    userId: z.string().uuid(),
+    createdAt: z.string(),
+    title: z.string(),
+    description: z.string(),
+    permissions: z.array(PermissionsModel),
+    assignedResources: ResourceAssignmentModel.nullable().transform(
+      value => value ?? { mode: '*' as const, projects: [] },
+    ),
+    firstCharacters: z.string(),
+    hash: z.string(),
+  })
+  .transform(record => ({
+    ...record,
+    // Only used in the context of authorization, we do not need
+    // to compute when querying a list of organization access tokens via the GraphQL API.
+    // Compared to organization access tokens, we also need to filter down the permissions based on the membership
+    resolveAuthorizationPolicyStatements(organizationMembership: OrganizationMembership) {
```

Comment:
In order to compute the authorization policy statements, we need to consider the organization membership, which is an input here.